src/AsmResolver.DotNet/Builder/PEImageBuildContext.cs

```diff
@@ -12,24 +12,43 @@ public class PEImageBuildContext
         /// </summary>
         public PEImageBuildContext()
         {
-            DiagnosticBag = new DiagnosticBag();
-        }
-        
+            ErrorListener = new DiagnosticBag();
+        }
+
         /// <summary>
         /// Creates a new build context.
         /// </summary>
-        /// <param name="diagnosticBag">The diagnostic bag to use.</param>
+        /// <param name="diagnosticBag">The diagnostic bag to use.</param>
+        [Obsolete]
         public PEImageBuildContext(DiagnosticBag diagnosticBag)
         {
-            DiagnosticBag = diagnosticBag ?? throw new ArgumentNullException(nameof(diagnosticBag));
-        }
-        
+            ErrorListener = diagnosticBag ?? throw new ArgumentNullException(nameof(diagnosticBag));
+        }
+
         /// <summary>
-        /// Gets the bag that collects all diagnostic information during the building process. 
+        /// Creates a new build context.
         /// </summary>
+        /// <param name="errorListener">The diagnostic bag to use.</param>
+        public PEImageBuildContext(IErrorListener errorListener)
+        {
+            ErrorListener = errorListener ?? throw new ArgumentNullException(nameof(errorListener));
+        }
+
+        /// <summary>
+        /// Gets the bag that collects all diagnostic information during the building process. 
+        /// </summary>
+        [Obsolete]
         public DiagnosticBag DiagnosticBag
+        {
+            get => (DiagnosticBag)ErrorListener;
+        }
+

```

Comment:
We may want to define this as a `DiagnosticBag?`, and use `as` instead of a direct cast to avoid casting exceptions. Also, let's notify consumers using the `Obsolete` CA that the `ErrorListener` is now preferred over the `DiagnosticBag`.